its/src/test/java/com/sonar/it/csharp/SdkTest.java

```diff
@@ -39,16 +39,36 @@ public class SdkTest {
   @TempDir
   private static Path temp;
 
+  @Test
+  void netCore31() throws IOException {
+    validateProject("NetCore31", 9, 14);
+  }
+
   @Test
   void net5() throws IOException {
-    validateProject("Net5");
+    validateProject("Net5", 3, 5);
+  }
+
+  @Test
+  void net6() throws IOException {
+    validateProject("Net6", 3, 5);
+  }
+
+  @Test
+  void net7() throws IOException {
+    validateProject("Net7", 3, 5);
+  }
+
+  @Test
+  void net8() throws IOException {
+    validateProject("Net8", 3, 5);
   }
 
-  void validateProject(String project) throws IOException {
+  void validateProject(String project, int issueLine, int lines) throws IOException {
```

Comment:
NetCore31 does not support top level statements, so this needs parametrization.

Reply:
What about not using top level statements? Make all of the projects DLL with something like
```
public class Sample
{
    public void Method()
    {
        // FIXME: ...
    }
}
```